test/org/sagebionetworks/bridge/services/EmailVerificationServiceTest.java

```diff
@@ -59,98 +71,163 @@ public void verifiedEmailTakesNoAction() {
+
+        Map<String, IdentityVerificationAttributes> map = Maps.newHashMap();
         map.put(EMAIL_ADDRESS, attributes);
         when(result.getVerificationAttributes()).thenReturn(null);
         when(sesClient.getIdentityVerificationAttributes(any())).thenReturn(result);
-        
+
         EmailVerificationStatus status = service.getEmailStatus(EMAIL_ADDRESS);
-        
+
         verify(sesClient).getIdentityVerificationAttributes(any());
         assertEquals(EmailVerificationStatus.UNVERIFIED, status);
+        
+        verify(cacheProvider).setString(eq(EMAIL_ADDRESS_KEY), eq("UNVERIFIED"), anyInt());        
     }
     
     @Test
     public void getEmailStatusVerificationStatusNull() {
         getCaptor = ArgumentCaptor.forClass(GetIdentityVerificationAttributesRequest.class);
-        
-        Map<String,IdentityVerificationAttributes> map = Maps.newHashMap();
+
+        Map<String, IdentityVerificationAttributes> map = Maps.newHashMap();
         map.put(EMAIL_ADDRESS, attributes);
         when(result.getVerificationAttributes()).thenReturn(map);
         when(attributes.getVerificationStatus()).thenReturn(null);
         when(sesClient.getIdentityVerificationAttributes(any())).thenReturn(result);
-        
+
         EmailVerificationStatus status = service.getEmailStatus(EMAIL_ADDRESS);
-        
+
         verify(sesClient).getIdentityVerificationAttributes(any());
         assertEquals(EmailVerificationStatus.UNVERIFIED, status);
+        
+        verify(cacheProvider).setString(eq(EMAIL_ADDRESS_KEY), eq("UNVERIFIED"), anyInt());        
     }
     
     @Test
     public void sendVerifyEmailRequest() {
         mockSession("Success");
         service.sendVerifyEmailRequest(EMAIL_ADDRESS);
-        
+
         verify(sesClient).verifyEmailIdentity(any());
+        
+        verify(cacheProvider).setString(eq(EMAIL_ADDRESS_KEY), eq("PENDING"), anyInt()); 
+    }
+    
+    @Test
+    public void isVerifiedAndCached() throws Exception {
+        when(cacheProvider.getString(EMAIL_ADDRESS_KEY)).thenReturn("VERIFIED");
+        assertTrue(service.isVerified(EMAIL_ADDRESS));
+    }
+    
+    @Test
+    public void isPendingAndCached() throws Exception {
+        when(cacheProvider.getString(EMAIL_ADDRESS_KEY)).thenReturn("PENDING");
+        assertFalse(service.isVerified(EMAIL_ADDRESS));
+    }
+    
+    @Test
+    public void isUnverifiedAndCached() throws Exception {
+        when(cacheProvider.getString(EMAIL_ADDRESS_KEY)).thenReturn("UNVERIFIED");
+        assertFalse(service.isVerified(EMAIL_ADDRESS));
+    }
+    
+    @Test
+    public void isVerifiedUncached() {
+        doReturn(EmailVerificationStatus.VERIFIED).when(service).getEmailStatus(EMAIL_ADDRESS);
+        
+        assertTrue(service.isVerified(EMAIL_ADDRESS));
+        
+        verify(cacheProvider).setString(eq(service.getVerifiedAddressKey(EMAIL_ADDRESS)),
+                eq("VERIFIED"), anyInt());
```

Comment:
Minor: Should we be using EMAIL_ADDRESS_KEY instead? (Here and in the two other tests below.)